types.go

```diff
@@ -76,47 +76,49 @@ func equals(a, b []string) bool {
 }
 
 // strPtrEqual returns true if the content of the provided string pointer are equal.
-func strPtrEqual(a, b *string) bool {
-	if a == b {
+func strPtrEqual(str1, str2 *string) bool {
+	if str1 == str2 {
 		return true
 	}
-	if a == nil || b == nil {
+	if str1 == nil || str2 == nil {
 		return false
 	}
-	return *a == *b
+	return *str1 == *str2
 }
 
 // Equal returns true if both blocks are functionally equivalent.
-func (d *Directive) Equal(a *Directive) bool {
-	if d == a {
+//
+//nolint:cyclop
+func (d *Directive) Equal(directive *Directive) bool {
+	if d == directive {
 		// same ptr, or both nil
 		return true
 	}
-	if d == nil || a == nil {
+	if d == nil || directive == nil {
```

Comment:
`directive` too close to struct name, since we are comparing one (`d`) against another, suggest `other`.